test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    constants: { AddressZero },
+    getContractAt,
+} = ethers;
+const { expect } = require('chai');
+const { getRandomBytes32 } = require('./utils');
+const { deployContract, deployAll } = require('../scripts/deploy');
+const { ITS_HUB_ADDRESS } = require('./constants');
+
+/**
+ * Convert bytes32 value to EIP-55 address
+ * @param {string} bytes - The bytes32 value
+ * @returns {string} The EIP-55 checksummed address
+ */
+function bytes32ToAddress(bytes) {
+    const addressHex = '0x' + bytes.slice(-40);
+    return ethers.utils.getAddress(addressHex);
+}
+
+describe('HyperliquidInterchainToken', () => {
+    let hyperliquidInterchainToken, hyperliquidInterchainTokenDeployer;
+
+    const name = 'HyperliquidToken';
+    const symbol = 'HLT';
+    const decimals = 18;
+    const mintAmount = 123;
+
+    let token;
+    let owner;
+    let user;
+    let provider;
+    let tokenAddress;
+    let slot0;
+    let deployerFromSlot0;
+    let deployerFromContract;
+
+    before(async () => {
+        const wallets = await ethers.getSigners();
+        owner = wallets[0];
+        user = wallets[1];
+
+        hyperliquidInterchainToken = await deployContract(owner, 'HyperliquidInterchainToken', [owner.address]);
```

Comment:
```suggestion
        token = await deployContract(owner, 'HyperliquidInterchainToken', [owner.address]);
```
just assign directly vs recreating the instance?